enhancements/security/x509-trust.md

```diff
@@ -0,0 +1,183 @@
+---
+title: x.509-trust
+authors:
+  - "@wking"
+reviewers:
+  - "@abhinavdahiya"
+  - "@bparees"
+  - "@dhansen"
+  - "@sdodson"
+approvers:
+  - TBD
+creation-date: 2019-11-14
+last-updated: 2019-11-14
+status: provisional
+see-also:
+  - "/enhancements/automated-service-ca-rotation.md"
+  - "/enhancements/kube-apiserver/certificates.md"
+  - "/enhancements/kube-apiserver/tls-config.md"
+  - "/enhancements/proxy/global-cluster-egress-proxy.md"
+---
+
+# X.509 Trust
+
+This enhancement provides a big-picture overview of X.509 trust management for cluster components communicating with external actors.
+This enhancement provides for both cluster-scoped defaults and per-component overrides.
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+Several cluster components may make TLS requests to external services.
+For example, [the cluster-version operator][cluster-version-operator] may request available updates, [the machine-API operator][machine-api-operator] may call platform APIs to manage machines, [CRI-O][cri-o] may contact container image registries when downloading images for pods, etc.
+Cluster components may also perform other X.509 verification besides TLS requests.
+For example, user mail processing components may perform [S/MIME][smime] verification.
+This enhancement provides a framework for managing X.509 trust for these situations.
+
+Currently the default trust for CRI-O, the kubelet's cloud provider, and other node-namespace egress is the node's X.509 certificate authority (CA) store, which depends on the operating system ([Red Hat Enterprise Linux CoreOS (RHCOS)][rhcos], etc.) and any subsequent configuration ([the machine-config operator][machine-config-operator], etc.).
+The default trust for containers like the machine-API operator is that container's CA store, which depends on the base layer ([Red Hat Universal Base Image (UBI)][ubi], etc.).
+Many external components will have X.509 certificates signed by one of those default CAs, but this is not always the case.
+
+This enhancement provides a mechanism for configuring additional or alternative trust bundles for all cluster services when communicating with external components whose X.509 certificates are not signed by a default CA.
```

Comment:
nit: `external components` we probably should explain external components.. does routes used internally count into this? api vs api-int etc.. ?

Reply:
> ... we probably should explain external components...

Yeah.  This is basically for everything that's not signed by a root CA living in the cluster.  Does that sound like a clear enough, achievable position?

Reply:
> ... we probably should explain external components...

I've pushed 3d92920 -> f451e19, rebasing on master and removing most `external` references.  I'm trying to say "use this mechanism if you do X.509 verification at all, unless you already have an existing mechanism like the Kube API cert setup for configuring your trust".